src/Scientist/Scientist.cs

```diff
@@ -27,14 +41,16 @@ public static IObservationPublisher ObservationPublisher
         /// <typeparam name="T">The return type of the experiment</typeparam>
         /// <param name="name">Name of the experiment</param>
         /// <param name="experiment">Experiment callback used to configure the experiment</param>
+       
         /// <returns>The value of the experiment's control function.</returns>
         [return: AllowNull]
-        public static T Science<T>(string name, Action<IExperiment<T>> experiment)
+        [MethodImpl(MethodImplOptions.NoInlining)] //So that we get the Source code CallerMemberName method name (may be lost when inlined in Release Mode).
+#pragma warning disable 1573
+        public static T Science<T>(string name, Action<IExperiment<T>> experiment, [CallerMemberName] string callingMethodName = "")
```

Comment:
`[CallerMemberName] string callingMethodName = ""` get the method from where Scienceing this was called.